### PR TITLE
[AIR] Time series AutoML with Ray AIR example

### PR DESCRIPTION
**Pull request by justinvyu**
Signed-off-by: Justin Yu <justinvyu@berkeley.edu>

## Why are these changes needed?

This PR shows a simple example of running an AutoML Time Series workload on AIR, porting over this [Ray Core example](https://docs.ray.io/en/master/ray-core/examples/automl_for_time_series.html). The goal is to use Tune to search over different autoregressive statistical forecasting models (using the library `statsforecast`) and returning the best model type as well as model parameters based on some metric.

See the notebook [here](https://colab.research.google.com/github/justinvyu/ray/blob/automl_air/doc/source/ray-air/examples/automl_with_ray_air.ipynb)!

Note: the notebook is still a work in progress and needs more text descriptions and polish.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-air/examples/automl_with_ray_air.ipynb [hunk 1]
```diff
@@ -0,0 +1,858 @@
...
+    "\n",
+    "The basic steps were:\n",
+    "\n",
+    "1. Define a set of autoregressive forecasting models to search over. For each model type, we also define a set of model parameters to search over.\n",
+    "2. Perform temporal cross-validation on each model and model parameter configuration in parallel.\n",
+    "3. Pick the best performing model as the output of the AutoML system.\n",
+    "\n",
+    "We see that these steps fit into the framework of a hyperparameter optimization problem that can be tackled with the [Ray AIR Tuner](air-tuner)!\n",
+    "\n",
+    "In this notebook, we will show how to create an AutoML system with Ray AIR. Compared to using Ray Core, we will leverage the higher-level Tuner API to define the model and hyperparameter search space, parallelize cross-validation of different models, and finally analyze results to pull the best-performing model for the time-series dataset. Similar to the Ray Core example, we will be using only one partition of the M5 dataset for this AutoML example."
```
Comment (Yard1) [inline doc/source/ray-air/examples/automl_with_ray_air.ipynb:26]: can we link to the ray core example?

**Review by Yard1 — APPROVED** [review]
Looks great, I don't have any major comments. Will give it a look one more time later to see if I catch any nits :)

File: doc/source/ray-air/examples/automl_with_ray_air.ipynb (hunk 1 above)
Comment (richardliaw) [reply doc/source/ray-air/examples/automl_with_ray_air.ipynb:26]: Can you also break this up into bullet points?

```
In this notebook, we will show how to:
  1. Create an AutoML system for time series forecasting
  2. Leverage the ...

File: doc/source/ray-air/examples/automl_with_ray_air.ipynb [hunk 2]
```diff
@@ -0,0 +1,858 @@
...
+    "    checkpoint = Checkpoint.from_dict(checkpoint_dict)\n",
+    "    session.report(results, checkpoint=checkpoint)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5cf6e512",
+   "metadata": {},
+   "source": [
+    "We also define a helper function used above, which calculates the metrics from the dataframe output of `StatsForecast.cross_validation`. For example, we will calculate the mean squared error between the model predictions and the actual observed data, averaged over all training splits. This metric gets reported to Tune as `mse_mean`, which is the metric we will use to define the best-performing model."
```
Comment (richardliaw) [inline doc/source/ray-air/examples/automl_with_ray_air.ipynb:342]: The flow here is a bit weird; you probably to
- move the docstring from the function below, out of the cell
- move this text above the previous cell
- use the text box above the previous cell to describe both functions, something like:

```
we will define two methods below:

- cross_validation, which does X
- compute_metrics_and_aggregate which is a helper function used in cross_validation and does ABC

**Review by richardliaw — COMMENTED** [review]
Looks pretty good, just some formatting improvements

**Review by richardliaw — APPROVED** [review]

**Pull request merged**